src/transformers/modeling_utils.py

```diff
@@ -2240,19 +2273,20 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                         if has_file(pretrained_model_name_or_path, TF2_WEIGHTS_NAME, **has_file_kwargs):
                             raise EnvironmentError(
```

Comment:
Suggestion: additional case that verifies whether a file without the variant exists.

                                f"{pretrained_model_name_or_path} does not appear to have a file named"
                                f" {WEIGHTS_NAME + variant_suffix} but there is a file without the variant"
                                f" named {WEIGHTS_NAME}. Use `variant=None` to load this model from those weights."